Blazor Hybrid dev tools topic (#25384)
dotnet/AspNetCore.Docs: aspnetcore/blazor/hybrid/developer-tools.md

---
title: Use browser developer tools with ASP.NET Core Blazor Hybrid
author: guardrex
description: Learn how to use browser developer tools with Blazor Hybrid apps.
monikerRange: '>= aspnetcore-6.0'
ms.author: riande
ms.custom: "mvc"
ms.date: 03/23/2022

> no-loc: ["Blazor Hybrid", Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]

Reviewer: Are these meta keywords? Can we add `.NET MAUI`?

Author: MT doesn't run on Blazor docs node. It's all HT ("human translation") here. However, it's possible that some or all of the topics would become MT at some point. Yes, I think it's a good idea to add it. I'll do that, but I'll do it on a separate PR. I'll note it on my tracking issue so that I don't lose this.

Author: Oh, sorry ... I missed the ❓ ...

> Are these meta keywords?

No, they're words/phrases to avoid for localization. .NET MAUI doesn't translate, but it would be safe to add it to the list, and I've made a note to do so.

Reviewer: Let's add `Mac Catalyst` as well, I've seen that unintentionally localized in the past.

Author: Got it ... I added "Mac Catalyst" to the UE tracking issue. It won't get lost.

> uid: blazor/hybrid/developer-tools

Reviewer: ```suggestion
uid: blazor/hybrid/accessing-browser-developer-tools
```

---
# Use browser developer tools with ASP.NET Core Blazor Hybrid

This article explains how to use [browser developer tools](https://developer.mozilla.org/docs/Glossary/Developer_Tools) with Blazor Hybrid apps.

[!INCLUDE[](~/blazor/includes/blazor-hybrid-preview-notice.md)]

## .NET MAUI Blazor Hybrid

To use developer tools with a .NET MAUI Blazor app, use the guidance in the following sections depending on the operating system used to run the app:

* [iOS](#ios)
* [Android](#android)

> 

Reviewer: @guardrex this wasn't covered as part of the original issue, but would you be able to add additional guidance for Windows & MacOS?

- Windows should just be a matter of CTRL+SHIFT+I within the BlazorWebView to open the dev tools.
- Mac should be an essentially identical process to ios.

Author: Yes, I'll do that first thing Thursday morning and ping u back.

### iOS

Use the following steps to use Safari developer tools with an iOS app. The following example shows how to inspect an unhandled exception in the **Web Inspector** console log:

1. Open desktop Safari.
1. Select the **Preferences** > **Advanced** > **Show Develop** menu item.

> 1. Run the .NET MAUI Blazor app in the Visual Studio emulator.

Reviewer: > Visual Studio emulator

Not sure what this may be referring to? Perhaps:

```suggestion
1. Run the .NET MAUI Blazor app in the iOS simulator.
```

> 1. Return to desktop Safari. Select **Develop** > **Simulator** > **0.0.0.0**. If multiple entries for **0.0.0.0** are present, select the entry that highlights the `BlazorWebView`. The `BlazorWebView` is highlighted in blue in the Visual Studio emulator when the correct **0.0.0.0** entry is selected.

Reviewer: ```suggestion
1. Return to desktop Safari. Select **Develop** > **Simulator** > **0.0.0.0**. If multiple entries for **0.0.0.0** are present, select the entry that highlights the `BlazorWebView`. The `BlazorWebView` is highlighted in blue in the iOS simulator when the correct **0.0.0.0** entry is selected.
```

   ![Safari Develop Simulator open showing two entries for "0.0.0.0" with the second entry selected because it highlights the BlazorWebView in the Visual Studio emulator UI.](~/blazor/hybrid/developer-tools/_static/ios.png)

> 1. The Simulator window opens, and the **Web Inspector** appears for the `BlazorWebView`.

Reviewer: The simulator should already be up and running at this time.

```suggestion
1. The **Web Inspector** window appears for the `BlazorWebView`.
```

> 1. Inspect the **Console** tab to see the console messages, which includes any exception messages generated by the framework or developer code:

Reviewer: I wonder if the current verbiage here may sound restrictive? Inspecting using the console tab is _one of_ the things you can do. The current phrasing sounds "instructive" as if it is the first/only thing you can do? Maybe I'm reading too much into it?

   ![Simulator window for a Blazor Hybrid app running on an emulated iPad mini](~/blazor/hybrid/developer-tools/_static/safari-debug-window.png)

### Android

Use the following steps to use browser developer tools with an Android app. The following example shows how to inspect an unhandled exception in the developer tools console log:

> 1. Start the Visual Studio Android emulator and navigate to an app page that uses a `BlazorWebView`.

Reviewer: ```suggestion
1. Start the Android emulator and navigate to an app page that uses a `BlazorWebView`.
```

1. Open Google Chrome or Microsoft Edge.
1. Navigate to `chrome://inspect/#devices` (Google Chrome) or `edge://inspect/#devices` (Microsoft Edge).
1. Select the **`inspect`** link button to open developer tools. The following example shows the **DevTools** page in Microsoft Edge:

   ![Microsoft Edge Devices showing the BlazorWebView's "inspect" link button to open developer tools.](~/blazor/hybrid/developer-tools/_static/android.png)

> 1. Inspect the **Console** tab to see the console messages, which includes any exception messages generated by the framework or developer code:

Reviewer: See above comment.
